crates/matrix-sdk/src/event_cache/linked_chunk/updates.rs

```diff
@@ -61,75 +61,110 @@ pub enum Update<Item, Gap> {
     /// A chunk has been removed.
     RemoveChunk(ChunkIdentifier),
 
-    /// Items are inserted inside a chunk of kind Items.
-    InsertItems {
-        /// [`Position`] of the items.
+    /// Items are pushed inside a chunk of kind Items.
+    PushItems {
+        /// The [`Position`] of the items.
+        ///
+        /// This value is given to prevent the need for position computations by
+        /// the update readers. Items are pushed, so the positions should be
+        /// incrementally computed from the previous items, which requires the
+        /// reading of the last previous item. With `at`, the update readers no
+        /// longer need to do so.
         at: Position,
 
         /// The items.
         items: Vec<Item>,
     },
 
-    /// A chunk of kind Items has been truncated.
-    TruncateItems {
-        /// The identifier of the chunk.
-        chunk: ChunkIdentifier,
-
-        /// The new length of the chunk.
-        length: usize,
+    /// The last items of a chunk have been detached, i.e. the chunk has been
+    /// truncated.
+    DetachLastItems {
+        /// The split position. Before this position (`[..position]`), items are
+        /// kept, from this position (`[position..]`), items are
```

Comment:
Are the examples in the parens supposed to be using range notation? I found the `[]` confusing, and thought it was maths intervals, and I was *just* about to request either `[..position[` or `]position..]` to make this clearer. Maybe we could remove the brackets?

Reply:
:+1: